### DISCUSSION TITLE
Scripts in _document.js.

### DISCUSSION THREAD

**Discussion by FReeeZzze** [vercel/next.js#11007] [Help]
How to solve the problem of an external script in _document? . The script is added after NextScript  but nevertheless does not work. The same script runs fine in React.

```
<body>
  <Main />
  <Preloader />
  <NextScript />
  <script type="text/javascript" dangerouslySetInnerHTML={this.WindowPreloader()} />
  <script type="text/javascript" src="https://widgets.radioheart.ru/static/js/manifest.js" />
  <script type="text/javascript" src="https://widgets.radioheart.ru/static/js/vendor.js" />
  <script type="text/javascript" src="https://widgets.radioheart.ru/static/js/app.js" />
  <script src="https://player.radioheart.ru/js/app.js" />
  <script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit" async defer />
</body>
```

**Comment by Timer** [comment]
I'm not sure I understand the question. Can you elaborate on the objective and problem more?

**Reply by westprophet** [reply]
I greet you. My colleague FReeeZzze and I had problems working with Next.js 9.3.0.

In the _documents.js file we insert <script> (CDN) which refers to an external source (radioheart.ru). We insert it before the </ body> tag, as the documentation for this script says.

This script does not work correctly in Next.js., but it works correctly in React.js, and on other sites.

If in _document.js before the problem <script> you do not specify <NextScript> then the problem script works correctly. But other scripts like Google recapcha stop working.

We tried different options and inserted this script into Head but it doesn’t help.

P.S I want to apologize for the translation, I used google translator to write this message. I hope I have explained the essence of our problem.

**Reply by westprophet** [reply]
We cannot find a problem why this external scripts do not work correctly in Next .js application

We are rebuild the site http://stellium.de which was created on React.js to Next .js.

**Reply by FReeeZzze** [reply]
> I'm not sure I understand the question. Can you elaborate on the objective and problem more?

`<script type = "text / javascript" src = "https://widgets.radioheart.ru/static/js/manifest.js" />` this script conflicts when I add it after NextScript. As a result, the content is not displayed, and the following messages are displayed in the console: [check this](https://ibb.co/8sK3QFx). When I delete the NextScript, there are no errors, but the content is not displayed, moreover, Google reaptcha does not work, so this is not the desired result.

**Comment by TyMick** [comment]
I'd try adding your scripts to [`_app.js`](https://nextjs.org/docs/advanced-features/custom-app) instead of [`_document.js`](https://nextjs.org/docs/advanced-features/custom-document). Does this `./pages/_app.js` fix the problem? After adding the logic for `this.WindowPreloader()`, that is.

```jsx
function MyApp({ Component, pageProps }) {
  return (
    <>
      <Component {...pageProps} />
      <script type="text/javascript" dangerouslySetInnerHTML={this.WindowPreloader()} />
      <script type="text/javascript" src="https://widgets.radioheart.ru/static/js/manifest.js" />
      <script type="text/javascript" src="https://widgets.radioheart.ru/static/js/vendor.js" />
      <script type="text/javascript" src="https://widgets.radioheart.ru/static/js/app.js" />
      <script src="https://player.radioheart.ru/js/app.js" />
      <script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit" async defer />
    </>
  );
}

export default MyApp;
```

**Reply by allandlobr** [reply]
Thank you 😄 . This topic is already 2 years old, but i got the same problem, i tried a way to load a script in the _document, because all of my pages need this script to be loaded. I loaded the script with the <Script> tag imported from 'next/script' in the MyApp function and it worked!

**Comment by 3lonious** [comment]
i dont have an _app.js file... my next app has never had one... can we please get a clear example with instructions form next/vercel as ive followed your git hun link with the google analytics tag example but it dosnt work after trying to copy the missing files over.

**Reply by TyMick** [reply]
Are you talking about the [with-google-analytics](https://github.com/vercel/next.js/tree/canary/examples/with-google-analytics) example? What's the issue you're encountering?

**Comment by ghost** [comment]
hello,  i am using the google analytics to integration . but my tracking id is not working or no data can be fetching
